### PR TITLE
Enable enter to advance

### PR DESCRIPTION
**Pull request by Tamir198**
## changes in create new proxy page

Adding the option to press enter to move on (only if next button is enabled) when adding a new proxy: 

image

Also, when adding a proxy now there is a debounced error message (shown to the user after a second) : 

image

---

## Changes on edit sniffer: 

- Allow save on enter
- Display error if the changes to the url are not formatted correctly
- Fix a bug where changing the name of the proxy would also change its subdomain.

image

---

### CODE REVIEW

File: client/src/pages/sniffers/AddSnifferPage.tsx [hunk 1]
```diff
@@ -124,17 +121,37 @@ function SimpleDomainComponent(props: {
...
 }) {
+  const [error, setError] = React.useState<string | null>(null);
+  const DEBOUNCE_TIME = 1000;
+
+  const handleDomainChange = (newValue: string) => {
+    props.onDomainChange(newValue);
+    debouncedSearch(newValue);
+  };
+
+  const debouncedSearch = React.useRef(
```
Comment (ypotter) [inline client/src/pages/sniffers/AddSnifferPage.tsx:132]: Why did you use useRef here?

File: client/src/pages/sniffers/useEnterKeyPress.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { useEffect } from "react";
+
+const ENTER_KEY = "Enter";
+
+export const useEnterKeyPress = (
```
Comment (ypotter) [inline client/src/pages/sniffers/useEnterKeyPress.tsx:6]: What would happen if in a single page there are 2 event listeners for ENTER? probably they both react to it isn't it?

**Review by ypotter — CHANGES_REQUESTED** [review]

File: client/src/utils/ValidateHttpUrl.tsx [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const validateHttpUrlFormat = (string: string) => {
+  const urlPattern = /^https?:\/\/[a-zA-Z0-9-]+(\.[a-zA-Z]{2,})+$/;
```
[the remark this answers is not in the document]
Comment (Tamir198) [reply client/src/utils/ValidateHttpUrl.tsx]: Added support for localhost, I chose to split the function into 2 separate regex (I thing its more readable)

File: client/src/pages/sniffers/AddSnifferPage.tsx [hunk 4]
```diff
@@ -267,7 +287,7 @@ export const AddSnifferPage = () => {
       name,
       downstreamUrl: domain,
       port: 80,
-      subdomain: `${name}-${subdomain}`,
+      subdomain: `${name.toLowerCase().replace(/\s/g, "-")}-${subdomain}`,
```
[the remark this answers is not in the document]
Comment (Tamir198) [reply client/src/pages/sniffers/AddSnifferPage.tsx]: Extracted

File: client/src/pages/sniffers/AddSnifferPage.tsx (hunk 1 above)
Comment (Tamir198) [reply client/src/pages/sniffers/AddSnifferPage.tsx]: I wanted to make sure that this function will not change across re renders (it's causing the error message to flicker). 
But that you mentioned it, no point in using `useRef`, I changed this into `useCallback` as I think it's a more appropriate way to handle this. 

Do you prefer a different solution?

File: client/src/pages/sniffers/useEnterKeyPress.tsx (hunk 2 above)
Comment (Tamir198) [reply client/src/pages/sniffers/useEnterKeyPress.tsx]: I think that you are right, I replaced this hook with a function + now we are using the `onKeyDown` property on inputs and text fields so in this way, the function triggered by <kbd>enter</kbd> will be associated with the relevant element element

File: client/src/pages/sniffers/EditSnifferModal.tsx [hunk 5]
```diff
@@ -68,6 +83,19 @@ export const EditSnifferModal = ({
       });
   }, [name, downstreamUrl, sniffer, showSnackbar, onClose]);
 
+  const [enterPressed, setEnterPressed] = React.useState(false);
+  // const handleKeyDown = (e) => {
```
Comment (ypotter) [inline client/src/pages/sniffers/EditSnifferModal.tsx:87]: Remove comments.

**Review by ypotter — CHANGES_REQUESTED** [review]

File: client/src/pages/sniffers/useEnterKeyPress.tsx (hunk 2 above)
Comment (ypotter) [reply client/src/pages/sniffers/useEnterKeyPress.tsx]: Much better!

File: client/src/pages/sniffers/AddSnifferPage.tsx (hunk 1 above)
Comment (ypotter) [reply client/src/pages/sniffers/AddSnifferPage.tsx]: Great!

**Review by ypotter — APPROVED** [review]

**Pull request merged**